plugin.rb

```diff
@@ -551,6 +564,17 @@ def event_ics(opts = {})
 
       render body: cal.to_ical, formats: [:ics], content_type: Mime::Type.lookup("text/calendar") unless performed?
     end
+
+    def current_user
+      if params.key?(USER_API_KEY) && params.key?(USER_API_CLIENT_ID)
+        # Logging in with user api keys normally only works by passing certain headers.
```

Comment:
@haveyaseen @fzngagan Could you explain why this is here? The calendar urls already work if you just add the user api key and user as query parameters. The discourse accepts API credentials as headers or as query parameters out of the box.

Reply:
What you said is true for the admin API keys. For *user* API keys, however, the header [`User-Api-Key` is required (see "Consuming the API")](https://meta.discourse.org/t/user-api-keys-specification/48536). You can check this in the [Discourse code that resolves `current_user`](https://github.com/discourse/discourse/blob/dc9110cc434b0c0f3d6bec8d992c6caea31e273c/lib/auth/default_current_user_provider.rb#L47) – the user API key is taken from `@env` (headers) while the admin API key one line below could either be in `@env` or `request` (query params).

I tried again just now. Without this block of code the entire feature does not work. Thinking about it, maybe the requirement for `params.key?(USER_API_CLIENT_ID)` should be dropped as Discourse doesn't need it. I'll go ahead and change that.